src/openrct2/management/Research.h

```diff
@@ -33,10 +33,10 @@ struct rct_research_item
     uint8_t category;
 
     bool IsInventedEndMarker() const;
-    bool IsRandomEndMarker() const;
-    bool IsUninventedEndMarker() const;
+    bool Equals(const ResearchItem* otherItem) const;
+    bool Exists() const;
 };
-assert_struct_size(rct_research_item, 5);
+assert_struct_size(ResearchItem, 5);
 #pragma pack(pop)
```

Comment:
If the S4 and S6 have convert routines then we no longer need to constrain the size of the struct.